tensorly/regression/cp_plsr.py

```diff
@@ -104,12 +104,10 @@ def fit(self, X, Y):
 
             for iter in range(self.n_iter_max):
                 Z = T.tensordot(X, comp_Y_factors_0, axes=((0,), (0,)))
-                if Z.ndim >= 2:
-                    Z_comp = tucker(Z, [1] * T.ndim(Z))[1]
-                else:
-                    Z_comp = [Z / T.norm(Z)]
+
                 for mode in range(1, X.ndim):  # First mode of Z is collapsed by the above tensordot call
-                    comp_X_factors[mode] = T.reshape(Z_comp[mode - 1], (-1,))
+                    Z_comp = svd_interface(unfold(Z, mode - 1), n_eigenvecs=1)[0]
```

Comment:
@aarmey I just found out that the change from `tucker()` to `svd_interface()` actually changes the result of decomposition. I'm not sure what `svd_interface()` does, but running a proper HOOI (which I assume `tucker()` does) is an integral part of this algorithm. Can you help me check on this function? Thank you!

Reply:
You're right. Since you only need one mode, take a look at partial_tucker. That should be the right choice.

Reply:
Hmm, it doesn't seem like there is too much performance difference between `tucker()` and `partial_tucker()` though, as the former pretty much just calls the latter.

Reply:
Performance is not the only consideration here. This code should be clear to read for others trying to understand the algorithm. Readability will also help others verify that the code is correct themselves.

Actually, reading more I'm surprised that the results are different when using the SVD of the unfolded tensor. This is exactly what Tucker uses for initialization, and my understanding is that the subsequent iteration within Tucker doesn't do all that much. Are you perhaps just seeing a sign difference in the results?

Reply:
I personally found the readability fine, as `partial_tucker()` actually takes more parameters that we don't need than `tucker()`, but if you find that better I can change it too as they are pretty much equivalent in our case.

For Tucker with ranks difference from the original shape, I think the subsequent iteration does make a difference?  As we don't have orthogonality like in SVD anymore in the tensor case.

Yes, the difference is nontrivial and more than a sign difference.

Reply:
Ok. Use `partial_tucker()` for now. I'll then look into this more.

Reply:
@cyrillustan, based on your investigation, change this to CP, then you can resolve this thread. :-)